### PR TITLE
Inner

### PR DESCRIPTION
**Pull request by sunjincheng121**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/expressions/call.scala [hunk 1]
```diff
@@ -205,7 +207,8 @@ case class OverCall(
 
     // check preceding is valid
     preceding match {
-      case _: CurrentRow | _: CurrentRange | _: UnboundedRow | _: UnboundedRange =>
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/expressions/call.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/expressions/call.scala [hunk 2]
```diff
@@ -221,7 +224,8 @@ case class OverCall(
 
     // check following is valid
     following match {
-      case _: CurrentRow | _: CurrentRange | _: UnboundedRow | _: UnboundedRange =>
+      case _: CurrentRow | _: CurrentRange
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/expressions/call.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/expressions/fieldExpression.scala [hunk 3]
```diff
@@ -120,7 +121,8 @@ case class UnresolvedAlias(child: Expression) extends UnaryExpression with Named
   override private[flink] lazy val valid = false
 }
 
-case class WindowReference(name: String, tpe: Option[TypeInformation[_]] = None) extends Attribute {
+case class WindowReference(name: String, tpe: Option[TypeInformation[_]] = None)
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/expressions/fieldExpression.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/logical/operators.scala [hunk 4]
```diff
@@ -59,7 +59,8 @@ case class Project(
             case ne: NamedExpression => ne
             case expr if !expr.valid => u
             case c @ Cast(ne: NamedExpression, tp) => Alias(c, s"${ne.name}-$tp")
-            case gcf: GetCompositeField => Alias(gcf, gcf.aliasName().getOrElse(s"_c$i"))
+            case gcf: GetCompositeField =>
+              Alias(gcf, gcf.aliasName().getOrElse(s"_c$i"))
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/logical/operators.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/logical/operators.scala [hunk 5]
```diff
@@ -205,9 +206,9 @@ case class Filter(condition: Expression, child: LogicalNode) extends UnaryNode {
 }
 
 case class Aggregate(
-    groupingExpressions: Seq[Expression],
-    aggregateExpressions: Seq[NamedExpression],
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/logical/operators.scala]: 缩进改错了

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/logical/operators.scala [hunk 6]
```diff
@@ -372,11 +373,11 @@ case class Intersect(left: LogicalNode, right: LogicalNode, all: Boolean) extend
 }
 
 case class Join(
-    left: LogicalNode,
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/logical/operators.scala]: 缩进改错了

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/LogicalWindowAggregateRule.scala [hunk 7]
```diff
@@ -146,6 +146,7 @@ abstract class LogicalWindowAggregateRule(ruleName: String)
       windowExpression: RexCall): RexNode
 
   /** translate the group window expression in to a Flink Table window. */
-  private[table] def translateWindowExpression(windowExpr: RexCall, rowType: RelDataType): Window
+  private[table] def translateWindowExpression(
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/LogicalWindowAggregateRule.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/WindowPropertiesRule.scala [hunk 8]
```diff
@@ -69,12 +69,16 @@ abstract class WindowPropertiesBaseRule(rulePredicate: RelOptRuleOperand, ruleNa
...
       case 'streamRowtime =>
         Seq(
-          NamedWindowProperty(propertyName(w, "rowtime"), RowtimeAttribute(w.aliasAttribute)),
-          NamedWindowProperty(propertyName(w, "proctime"), ProctimeAttribute(w.aliasAttribute)))
+          NamedWindowProperty(
+            propertyName(w, "rowtime"), RowtimeAttribute(w.aliasAttribute)),
+          NamedWindowProperty(
+            propertyName(w, "proctime"), ProctimeAttribute(w.aliasAttribute)))
       case 'streamProctime =>
-        Seq(NamedWindowProperty(propertyName(w, "proctime"), ProctimeAttribute(w.aliasAttribute)))
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/WindowPropertiesRule.scala]: 除了package rename，都是unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/validate/FunctionCatalog.scala [hunk 9]
```diff
@@ -267,6 +323,8 @@ object FunctionCatalog {
     "dateTimePlus" -> classOf[Plus],
     "dateFormat" -> classOf[DateFormat],
     "timestampDiff" -> classOf[TimestampDiff],
+    "temporalFloor" -> classOf[TemporalFloor],
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/validate/FunctionCatalog.scala]: 为什么需要加这2个？

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/GroupWindowTest.scala [hunk 10]
```diff
@@ -674,7 +674,8 @@ class GroupWindowTest extends TableTestBase {
           "DataStreamGroupWindowAggregate",
           streamTableNode(0),
           term("groupBy", "string, int2, int3"),
-          term("window", SlidingGroupWindow(WindowReference("w"), 'proctime,  2.rows, 1.rows)),
+          term("window",
+            SlidingGroupWindow(WindowReference("w"), 'proctime,  2.rows, 1.rows)),
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/GroupWindowTest.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/TemporalTableJoinTest.scala [hunk 11]
```diff
@@ -152,13 +152,14 @@ class TemporalTableJoinTest extends TableTestBase {
 
   @Test
   def testProcessingTimeIndicatorVersion(): Unit = {
-    assertRatesFunction(proctimeRatesHistory.getSchema, proctimeRates, true)
+    assertRatesFunction(proctimeRatesHistory.getSchema,
+      proctimeRates.asInstanceOf[TemporalTableFunction], true)
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/TemporalTableJoinTest.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/TemporalTableJoinTest.scala [hunk 12]
```diff
@@ -152,13 +152,14 @@ class TemporalTableJoinTest extends TableTestBase {
...
   }
 
   @Test
   def testValidStringFieldReference(): Unit = {
     val rates = ratesHistory.createTemporalTableFunction("rowtime", "currency")
-    assertRatesFunction(ratesHistory.getSchema, rates)
+    assertRatesFunction(ratesHistory.getSchema, rates.asInstanceOf[TemporalTableFunction])
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/TemporalTableJoinTest.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/validation/CorrelateValidationTest.scala [hunk 13]
```diff
@@ -77,12 +77,6 @@ class CorrelateValidationTest extends TableTestBase {
       "Table functions can only be used in table.joinLateral() and table.leftOuterJoinLateral()."
     )
 
-    // table function call filter
-    expectExceptionThrown(
-      func1('c).filter('f0 === "?"),
-      "Table functions can only be used in table.joinLateral() and table.leftOuterJoinLateral()."
-    )
-
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/validation/CorrelateValidationTest.scala]: 可以加回来了，source/sink那个好像可以把这个解了，现在可以过了。

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/TimeIndicatorConversionTest.scala [hunk 14]
```diff
@@ -22,7 +22,8 @@ import java.sql.Timestamp
 
 import org.apache.flink.api.scala._
 import org.apache.flink.table.api.scala._
-import org.apache.flink.table.expressions.{TimeIntervalUnit, WindowReference}
+import org.apache.flink.table.expressions.TimeIntervalUnit
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/TimeIndicatorConversionTest.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ArrayTypeTest.scala [hunk 15]
```diff
@@ -54,13 +54,13 @@ class ArrayTypeTest extends ArrayTypeTestBase {
       "[2, 9]")
 
     testAllApis(
-      array(Null(Types.INT), 1),
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ArrayTypeTest.scala]: unnecessary change as Null Expression has been removed

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/MapTypeTest.scala [hunk 16]
```diff
@@ -54,7 +54,7 @@ class MapTypeTest extends MapTypeTestBase {
       "{2=2, 3=9}")
 
     testAllApis(
-      map(1, Null(Types.INT)),
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/MapTypeTest.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/RowTypeTest.scala [hunk 17]
```diff
@@ -52,7 +52,7 @@ class RowTypeTest extends RowTypeTestBase {
       "1985-04-11,0.1,[1, 2, 3],{foo=bar},1,true") // string flatten
 
     testAllApis(
-      row(1 + 1, 2 * 3, Null(Types.STRING)),
+      row(1 + 1, 2 * 3, ApiNull(Types.STRING)),
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/RowTypeTest.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ScalarFunctionsTest.scala [hunk 18]
```diff
@@ -115,13 +116,13 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "null")
 
     testAllApis(
-      'f0.replace(Null(Types.STRING), ""),
+      'f0.replace(ApiNull(Types.STRING), ""),
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ScalarFunctionsTest.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ScalarFunctionsTest.scala [hunk 19]
```diff
@@ -1965,12 +1966,6 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "LOG(10, 100)",
       "2.0"
     )
-
-    testTableApi(
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ScalarFunctionsTest.scala]: why removed?

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ScalarOperatorsTest.scala [hunk 20]
```diff
@@ -308,7 +309,7 @@ class ScalarOperatorsTest extends ScalarOperatorsTestBase {
     )
 
     testTableApi(
-      'f10.in("FAIL", "FAIL", Null(Types.STRING)),
+      'f10.in("FAIL", "FAIL", ApiNull(Types.STRING)),
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ScalarOperatorsTest.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/UserDefinedScalarFunctionTest.scala [hunk 21]
```diff
@@ -121,25 +121,25 @@ class UserDefinedScalarFunctionTest extends ExpressionTestBase {
   @Test
   def testNullableParameters(): Unit = {
     testAllApis(
-      Func3(Null(INT_TYPE_INFO), Null(STRING_TYPE_INFO)),
+      Func3(ApiNull(INT_TYPE_INFO), ApiNull(STRING_TYPE_INFO)),
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/UserDefinedScalarFunctionTest.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/runtime/stream/TimeAttributesITCase.scala [hunk 22]
```diff
@@ -557,15 +558,26 @@ class TimeAttributesITCase extends AbstractTestBase {
 
     // Java expressions
 
+    val javaRegitsterMethod = tEnv.getClass.getSuperclass.getDeclaredMethod(
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/runtime/stream/TimeAttributesITCase.scala]: javaRegitsterMethod -> javaRegisterMethod

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/runtime/stream/sql/SqlITCase.scala [hunk 23]
```diff
@@ -230,7 +230,7 @@ class SqlITCase extends StreamingWithStateTestBase {
     val tEnv = StreamTableEnvironment.create(env)
     StreamITCase.clear
 
-    val sqlQuery = "SELECT b, COUNT(a) FROM MyTable GROUP BY b"
+    val sqlQuery = "SELECT b, COUNT(a) as x FROM MyTable GROUP BY b"
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/runtime/stream/sql/SqlITCase.scala]: unnecessary change

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/runtime/stream/table/AggregateITCase.scala [hunk 24]
```diff
@@ -51,13 +50,13 @@ class AggregateITCase extends StreamingWithStateTestBase {
     val testAgg = new DataViewTestAgg
     val t = StreamTestData.get5TupleDataStream(env).toTable(tEnv, 'a, 'b, 'c, 'd, 'e)
       .groupBy('e)
-      .select('e, testAgg.distinct('d, 'e))
+      .select('e, testAgg.distinct('d, 'e), 'e.sum)
```
Comment (dianfu) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/runtime/stream/table/AggregateITCase.scala]: unnecessary change

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala [hunk 25]
```diff
@@ -669,18 +669,21 @@ abstract class StreamTableEnvironment(
       case (RowtimeAttribute(UnresolvedFieldReference(name)), idx) =>
         extractRowtime(idx, name, None)
 
-      case (Alias(RowtimeAttribute(UnresolvedFieldReference(origName)), name, _), idx) =>
+      case (Alias(
+      RowtimeAttribute(UnresolvedFieldReference(origName)), name, _), idx) =>
```
[the remark this answers is not in the document]
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala]: 只添加了换行 +1

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/StreamTableEnvironment.scala [hunk 26]
```diff
@@ -124,10 +126,14 @@ class StreamTableEnvironment @deprecated(
     * @param fields The field names of the registered table.
     * @tparam T The type of the [[DataStream]] to register.
     */
-  def registerDataStream[T](name: String, dataStream: DataStream[T], fields: Expression*): Unit = {
+  def registerDataStream[T](
+      name: String,
+      dataStream: DataStream[T],
+      fields: Expression*): Unit = {
```
[the remark this answers is not in the document]
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/StreamTableEnvironment.scala]: 变化了2次相当于没有变化 +1

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/expressionDsl.scala [hunk 27]
```diff
@@ -488,13 +490,13 @@ trait ImplicitExpressionOperations {
   def trim(
       removeLeading: Boolean = true,
       removeTrailing: Boolean = true,
-      character: Expression = TrimConstants.TRIM_DEFAULT_CHAR) = {
+      character: Expression = new ELiteral(" ")) = {
```
[the remark this answers is not in the document]
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/expressionDsl.scala]: 不确定，这只是个别名，你有更好的名字吗？

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/table.scala [hunk 28]
```diff
@@ -187,8 +200,8 @@ class Table(
     */
   def createTemporalTableFunction(
       timeAttribute: String,
-      primaryKey: String): TemporalTableFunction = {
```
[the remark this answers is not in the document]
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/table.scala]: 这个包有依赖，TemporalTableFunction在planner里面

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/table.scala [hunk 29]
```diff
@@ -958,35 +1007,44 @@ class Table(
...
-    orderBy(parsedFields: _*)
+  def orderBy(fields: Expression*): Table = {
+    val parsedFields = fields.map(_.accept(new DefaultExpressionVisitor))
+    orderByInternal(parsedFields: _*)
+  }
+
+  private def orderByInternal(fields: PlannerExpression*): Table = {
+    val order: Seq[Ordering] = fields.map {
+      case o: Ordering => o
+      case asc: Call if "asc".equals(asc.functionName) =>
```
[the remark this answers is not in the document]
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/table.scala]: 如果改成protected还是 PublicEvolving吗？并且我们允许用户new FunctionDefinition吧？
用equalsIgnoreCase，function名字不区分大小写。？

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/expressions/ExpressionUtils.scala [hunk 30]
```diff
@@ -0,0 +1,152 @@
...
+
+import org.apache.flink.api.common.typeinfo.{BasicTypeInfo, TypeInformation}
+import org.apache.flink.streaming.api.windowing.time.{Time => FlinkTime}
+import org.apache.flink.table.api.ValidationException
+import org.apache.flink.table.calcite.FlinkTypeFactory
+import org.apache.flink.table.typeutils.{RowIntervalTypeInfo, TimeIntervalTypeInfo}
+
+object ExpressionUtils {
+  /**
+    * Retrieve result type of given Expression.
```
[the remark this answers is not in the document]
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/expressions/ExpressionUtils.scala]: 为啥要这样变化？

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/WindowPropertiesRule.scala (hunk 8 above)
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/WindowPropertiesRule.scala]: 超长了

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/validate/FunctionCatalog.scala (hunk 9 above)
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/validate/FunctionCatalog.scala]: 这是以前拉下的吧？

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/TemporalTableJoinTest.scala (hunk 11 above)
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/TemporalTableJoinTest.scala]: 超长了

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/TimeIndicatorConversionTest.scala (hunk 14 above)
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/TimeIndicatorConversionTest.scala]: 不是同一个包了

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ScalarFunctionsTest.scala (hunk 19 above)
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/plan/expressions/ScalarFunctionsTest.scala]: 同问，这个为啥去掉了？

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/table.scala (hunk 28 above)
[the remark this answers is not in the document]
Comment (dianfu) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/table.scala]: 嗯，有道理，+1

File: flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/FunctionType.java [hunk 31]
```diff
@@ -0,0 +1,30 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.expressions;
+
+/**
+ * Function type.
```
Comment (sunjincheng121) [inline flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/FunctionType.java]: Types for identify the type of function.

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/DataSetConversions.scala [hunk 32]
```diff
@@ -21,6 +21,7 @@ import org.apache.flink.api.common.typeinfo.TypeInformation
 import org.apache.flink.api.scala._
 import org.apache.flink.table.api.Table
 import org.apache.flink.table.expressions.Expression
+import org.apache.flink.table.plan.expressions.PlannerExpression
```
Comment (sunjincheng121) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/DataSetConversions.scala]: useless.

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/DataStreamConversions.scala [hunk 33]
```diff
@@ -19,8 +19,9 @@ package org.apache.flink.table.api.scala
 
 import org.apache.flink.api.common.typeinfo.TypeInformation
 import org.apache.flink.table.api.Table
-import org.apache.flink.table.expressions.Expression
+import org.apache.flink.table.plan.expressions.PlannerExpression
```
Comment (sunjincheng121) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/scala/DataStreamConversions.scala]: useless...

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/LogicalWindowAggregateRule.scala (hunk 7 above)
Comment (sunjincheng121) [reply flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/plan/rules/common/LogicalWindowAggregateRule.scala]: 超长了

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/sources/FilterableTableSource.scala [hunk 34]
```diff
@@ -21,6 +21,7 @@ package org.apache.flink.table.sources
 import java.util.{List => JList}
 
 import org.apache.flink.table.expressions.Expression
+
```
Comment (sunjincheng121) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/sources/FilterableTableSource.scala]: revert

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/CalcTest.scala [hunk 35]
```diff
@@ -83,10 +80,7 @@ class CalcTest extends TableTestBase {
           ),
           term("groupBy", "b"),
           term("window",
-            TumblingGroupWindow(
-              WindowReference("w"),
-              'rowtime,
-              5.millis)),
+            TumblingGroupWindow(WindowReference("w"), 'rowtime, 5.millis)),
```
Comment (sunjincheng121) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/stream/table/CalcTest.scala]: revert
